google/cloud/spanner_v1/transaction.py

```diff
@@ -142,6 +160,8 @@ def commit(self, return_commit_stats=False, request_options=None):
         :raises ValueError: if there are no mutations to commit.
         """
         self._check_state()
+        if self._transaction_id is None:
+            self.begin()
```

Comment:
This has been added to handle mutation case. 

Consider a transaction having only mutations operation in it, in this case when we reach to the commit operation, there will not be any transaction id present, and commit operation can not be inlined. 

As per the design document for ILB we need to make an explicit begin rpc call in case transaction contains only mutation operations.

Reply:
@larke I have added one more condition to begin transaction only if there is any mutation present. Please let me know if this looks ok ?

Reply:
Yes, the added mutations check makes it much clearer :+1:  LGTM